### PR TITLE
{phys}[intel/2021b] Yambo v5.1.2

### PR DESCRIPTION
**Pull request by ItIsI-Orient**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/y/Yambo/Yambo-5.2-intel-2021b.eb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+source_urls = ['https://github.com/yambo-code/yambo/archive/']
+sources = [
+    {   'filename': '%s.tar.gz' % local_commit,
+        'download_filename': 'yambo-%(version)s.tar.gz',
+        'extract_cmd': 'mkdir yambo-%(version)s && tar --strip-components=1 -xzf %s -C yambo-%(version)s' 
+    },  # Yambo
+    {
+        'source_urls': ['https://github.com/yambo-code/yambo-libraries/archive/'],
+        'filename': '1.0.tar.gz',
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yambo/Yambo-5.2-intel-2021b.eb:27]:
```python
local_yambo_libs_ver = '1.0'
...
    'download_file': '%s.tar.gz' % local_yambo_libs_ver,
    'filename': 'yamlbo-libraries-%s.tar.gz' % local_yambo_libs_ver,

File: easybuild/easyconfigs/y/Yambo/Yambo-5.2-intel-2021b.eb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        'extract_cmd': 'mkdir yambo-%(version)s/lib/yambo/ && tar --strip-components=1 -xzf %s -C yambo-%(version)s/lib/yambo/',
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yambo/Yambo-5.2-intel-2021b.eb:28]: use `local_extract_cmd`?

File: easybuild/easyconfigs/y/Yambo/Yambo-5.2-intel-2021b.eb [hunk 3]
```diff
@@ -0,0 +1,82 @@
+easyblock = 'MakeCp'
+
+name = 'Yambo'
+version = '5.2'
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yambo/Yambo-5.2-intel-2021b.eb:4]: not an official 5.2 release, so use `5.2.dev-<date-of-commit>` as version?

And maybe use `%(version_major_minor)s` below

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/y/Yambo/Yambo-5.2.dev-20230512-intel-2021b.eb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        'source_urls': ['https://github.com/yambo-code/yambo/files/962173/'],
+        'filename': 'iotk-y1.2.2.tar.gz',
+        'extract_cmd': 'cp %s yambo-%(version)s/lib/archive/',
+    },
+]
+checksums = [
+    ('4a993b86543fec7c5fa79838ab699a9f4432a215f63eca537c0819335ab9981c',
+    'f25d86101cc9cead9f64f05181aab1f06eac8528fea5ba5c586b2b9512cb8492',
+    'c0a4eb19f3e885d83d7afa52eb90658fba7cb1cb6e66049866a98dcc980de543')
```
Comment (boegel) [inline easybuild/easyconfigs/y/Yambo/Yambo-5.2.dev-20230512-intel-2021b.eb:42]: @ItIsI-Orient This syntax is definitely incorrect. You have a list with a single element, which is a tuple with 3 values.

Although that's supported, it doesn't specify what you probably intend to: if gives 3 different valid checksums for the 1st source file (and hence there are no checksums at all provided for the 2nd and 3rd source).

This should work (a list with 3 elements):

```python
checksums = [
    '4a993b86543fec7c5fa79838ab699a9f4432a215f63eca537c0819335ab9981c',
    'f25d86101cc9cead9f64f05181aab1f06eac8528fea5ba5c586b2b9512cb8492',
    'c0a4eb19f3e885d83d7afa52eb90658fba7cb1cb6e66049866a98dcc980de543',
]
```

This should also work:

```python
checksums = [
    {'yambo-5.2.dev-20230512.tar.gz': '4a993b86543fec7c5fa79838ab699a9f4432a215f63eca537c0819335ab9981c'},
    {'yambo-libraries-1.0.tar.gz': 'f25d86101cc9cead9f64f05181aab1f06eac8528fea5ba5c586b2b9512cb8492'},
    {'iotk-y1.2.2.tar.gz': 'c0a4eb19f3e885d83d7afa52eb90658fba7cb1cb6e66049866a98dcc980de543'},
]
```

Note that the keys must match with the filenames of the sources as they get stored in the source path, or you'll hit a confusing error like:

```
Invalid checksum spec 'None', should be a string (MD5) or 2-tuple (type, value).
```

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/y/Yambo/Yambo-5.2.dev-20230512-intel-2021b.eb (hunk 4 above)
Comment (boegel) [reply easybuild/easyconfigs/y/Yambo/Yambo-5.2.dev-20230512-intel-2021b.eb:42]: The latter is actually a bug, see also https://github.com/easybuilders/easybuild-framework/pull/4150

**Comment by ItIsI-Orient** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3133.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/bce4a58c9a651dc799a0d54d94cf8cfe for a full test report.

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/y/Yambo/Yambo-5.2.dev-20230512-intel-2021b.eb [hunk 5]
```diff
@@ -0,0 +1,85 @@
+easyblock = 'MakeCp'
+
+name = 'Yambo'
+version = '5.2.dev-20230512'
```
[the remark this answers is not in the document]
Comment (boegel) [reply easybuild/easyconfigs/y/Yambo/Yambo-5.2.dev-20230512-intel-2021b.eb:4]: @ItIsI-Orient I started look into this myself, but didn't fully finish it yet, WIP easyconfig available at [Yambo-5.1.2-intel-2021b.eb](https://github.com/vscentrum/vsc-software-stack/blob/wip/16_Yambo/Yambo-5.1.2-intel-2021b.eb)

**Comment by ItIsI-Orient** [comment]
Test report by @ItIsI-Orient
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node4008.donphan.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 530.30.02, Python 3.6.8
See https://gist.github.com/ItIsI-Orient/7755bd59de47614d6750a83fe3042920 for a full test report.

**Comment by ItIsI-Orient** [comment]
@boegelbot please test @ generoso

**Comment by ItIsI-Orient** [comment]
Test report by @ItIsI-Orient
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node4008.donphan.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz, 1 x NVIDIA NVIDIA A2, 530.30.02, Python 3.6.8
See https://gist.github.com/ItIsI-Orient/d252d22fbb455b3593ec42c866d3cb13 for a full test report.

**Comment by ItIsI-Orient** [comment]
@boegelbot please test @ jsc-zen2

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3119.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/a2d9dc13b70fe0cd9304092ba5f3a011 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3119.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/e6943d27bb5c7625d16cc90de0d11398 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @ItIsI-Orient!

**Pull request merged**